hyperon_das/query_engines/local_query_engine.py

```diff
@@ -158,16 +164,20 @@ def _process_link(self, query: dict) -> List[dict]:
 
         return result
 
-    def _generate_target_handles(self, targets: List[Dict[str, Any]]) -> List[str]:
-        targets_hash = []
+    def _generate_target_handles(self, targets: List[Dict[str, Any]]) -> list[str]:
+        targets_hash: list[str] = []
         for target in targets:
+            handle: str | list[str] | None = None
             if target["atom_type"] == "node":
                 handle = self.local_backend.node_handle(target["type"], target["name"])
             elif target["atom_type"] == "link":
-                handle = self._generate_target_handles(target)
+                handle = self._generate_target_handles([target])  # TODO: check if this is correct
             elif target["atom_type"] == "variable":
                 handle = WILDCARD
-            targets_hash.append(handle)
+            if handle and isinstance(handle, str):  # TODO: check if this is correct
+                targets_hash.append(handle)
+            elif handle and isinstance(handle, list):  # TODO: check if this is correct
```

Comment:
OK, will revisit this.

Reply:
Just checked this out again.

`_generate_target_handles` is supposed to return a `list[str]`. Even because its return is used to call `AtomDB.get_matched_links(<link-type>, target_handles)` (which expects a `list[str]` in `target_handles`).

If I revert my change, `_generate_target_handles` could return `list[str | list[str] | list[list[str]] | ...]` due to this `elif` (line 174 - which calls `_generate_target_handles` recursively):
```python
            elif target["atom_type"] == "link":
                handle = self._generate_target_handles([target])
```

Example of a possible return with the previous code: `['a', 'b', ['c', 'd'], 'e', [['f', 'g', 'h'], ['i',  'j']], 'k']`
My change flattens the return, resulting into: `['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i',  'j', 'k']`

Note: the behavior of `AtomDB.get_matched_links(...)` was not changed by my recent refactoring.

Reply:
> Example of a possible return with the previous code: ['a', 'b', ['c', 'd'], 'e', [['f', 'g', 'h'], ['i',  'j']], 'k']

This is correct. The nested lists are because we may have links pointing to links pointing to links, etc. 

If `get_matched_link()` is not considering the possibility of this method returning a nested list structure them we probably have a bug when the caller passes a query with a nested link in it.

Reply:
Changes on `_generate_target_handles` partially reverted. I will create a ticket to revisit this case in the future.